### ISSUE TITLE
A11y Audit - Screen Reader Testing

### ISSUE THREAD

**Issue by StephDriver** [BirkbeckCTP/janeway#4194]
# Context
Manual Testing of a sample of pages for screen reader capacity.  This is for functionality, not user experience.  The aim is to identify 'low hanging fruit' issues which can be addressed before future more exhaustive testing including direct user feedback.

Testing is non-exhaustive.  

[edit] It was going to include multiple OS and browsers, and several different journals, with the first tests on iOS Voiceover  on Safari because that's the one I know the best so I am more likely to catch such issues and reduces lead time on training on a new system.  Early results from this after 2 sample articles show that the many of the errors identified are then clearly visible in the code.  They are not 'I wonder how a different screen reader would handle this' but rather 'this heading should be an H5 instead of an H4' which would affect all users/screenreader softwares.  So this exercise has been revised:  it will be a representative sampling just with iOS Voiceover, then after the identified issues have been addressed, we will have another go at the original plan for testing with multiple different screen readers, browsers etc.

## Approach
- [x] test some pages manually
- [x] collate findings
- [x] define issues arising
- [x] place issues in https://github.com/BirkbeckCTP/janeway/milestone/38 or or later phases (e.g. for user content issues)
- [x] cross link issues back to these results
- [x] discuss with development team.

**Comment by StephDriver** [comment]
## [Sample 1 - Article](https://ijwwe.uwp.co.uk/article/id/9711/)
notes from traversing from top to bottom of page

- Journal banner image is read out correctly as the name of journal and identified as a landmark
- hamburger menu is incorrectly identified as "9711 link". Menu can be entered, which opens up the drop down options.
- Search icon is not identified simply read as 'link' however when activated, then the next is correctly identified as 'search field'
- Research Articles is identified as Main and Landmark, [should this be singular not plural?] but not as any kind of heading.
- The article name is correctly read out, but as a heading level 2.  There has been no heading level 1 before this.
- Author is not a heading of any kind.
- Author's name and affiliation is correctly read.
- The icon after author's name is not identified, simply 'link' `<a itemprop="email" href="mailto:lloyd@lemoyne.edu" class="fa fa-envelope email-link"></a> `
-

**Comment by StephDriver** [comment]
## [Sample 2 - Article](https://www.glossa-journal.org/article/id/10082/) in Material Theme

### using iOS with Safari and Voiceover
traversing from top of page to bottom

1. :x: "Glossa al of general linguistics, link, image, navigation, landmark" 
2. :x: home, link, list start - but nothing visible on the page and is a duplicate of the link above.
3. :x: Visually, at phone screen width there is a hamburger menu, but to the screen reader, no such thing exists, but instead the whole uncollapsed menu is read out despite being hidden.  These menu items are not identified as a menu or as navigation.
4. :x: items in the menu (articles, submissions, journal policies, about) are read followed by "arrow underscore drop underscore down" and identified as links. `<i class="material-icons right">arrow_drop_down</i>`
6. Search, link
7. Login, link
8. Register, link, list end
9. :x: "Menu, link, end, navigation, end, banner". This is the hamburger menu, and when activated the menu does pop up, but you can't navigate the list, as you already have - what pops up visually is the list that I've just described above, but it has already been navigated, when advancing one reaches the article title instead.  Further, there is no way now to dismiss this menu without refreshing the whole page, activating the menu link again does nothing.
10. :x: Article title is read out correctly but identified as an image - so is the alt for the image, then when moving on, one gets "Research Article" followed by the title again (!) none of which is identified as a heading (should be H1).

**Comment by StephDriver** [comment]
[Sample 3 - Article ](https://politicalphilosophyjournal.org/article/id/15481/) in Clean Theme

**Comment by StephDriver** [comment]
Sample 4 - Article in OLH theme (full article displayed on page)

**Comment by StephDriver** [comment]
Sample 5 - Non-article pages

**Comment by StephDriver** [comment]
## Other Observations

1. When navigating away from Sample 1 page, e.g. following a link to another page, the back button ends up returning to the article search page (from which this article was chosen) and not to the article itself. This is highly confusing.
3. headings seem to be split when ever there is another tag inside them. I'm not sure whether this is a peculiarity of voiceover or a problem with our HTML. Need to investigate further. It is highly annoying! Because in this article there is a lot of use of <em> inside headings due to the subject matter, and so single headings end up being split as if three different by consecutive headings.
4. despite both being H2, the "Download" heading is larger text than the "1 Introduction" heading. Is there consistency in how these styles are being applied, because otherwise they have the same font, and as font size is indicative of heading level, having H2 of inconsistent size is confusing.
5. is the use of `[ ]` within quotations enough to indicate to screen reader users that such words have been inserted into the original quote? Would a user expect to change their punctuation settings when this detail is needed, or would they want quotations marked up in such a way to make it more audible regardless of punctuation settings?
6. using small identifying marks such as 27 to 27' to differentiate between two items is difficult visually as well as for screen readers as described above.
7. my experience had no change in tone for `<em>` or `<strong>` but as the html appears to be correct, this may be an issue with my use of voiceover or my settings, so need to look at whether I have same problem with other sites.
8. in the footer where it is Published by Open Library of Humanities should that be Published by the Open Library of Humanities
9. navigation menu - Login and Register appear side by side, but all other links have their own line
10.  the way the author line sits across the article image in narrow screens is awkward - cuts across the eyeline of one of hte face images on the right.
11. further to item 14, "contact" text itself is not formatted, such that it default to the user's browser settings, possibly the only text on the page that doesn't have a font set.  If user has a very different default font to the theme, this looks very odd!
12. Roman numerals are not just a screen reader A11y issue.  There doesn't seem to be an accepted solution, with general advice being to limit their use. `Avoid the use of Roman Numerals and unfamiliar symbols in text where possible.` [WC3 guidance](https://www.w3.org/WAI/WCAG2/supplemental/patterns/o3p06-format-punctuation/)
13. is `lang` markup needed for Latin phrases in common use within English?  
14.  review use of `<blockquote>` and `<q>` tags, and their `cite` attributes.
15. abbreviations such as `e.g.` and `i.e.` are automatically read out as `for example` and `such that` respectively.  While not a problem here, might be worth considering highlighting this behaviour to editors.
16.  I am struggling to get the internal links back and forth to endnotes to work well.  Need to find out whether this is my inexperience with screen readers or a functional issue with the code.
17. should the 'bullet' in lists be read out? or marked as decorative?
18. decimals read out as dot not point?
19. abbreviations that can be read as words are read as words - need to mark them up as abbreviations where-ever used.
20. are we differentiating in terms of language between Englishes, and if so, does it affect voiceover?
21. something that has become annoying very fast has been navigating by headings, when there are lots of subheadings - need to learn whether this is about my knowledge - i.e. there is a simple way to navigate to the next same level heading, or whether this is purely about page navigability in which case some way to move between same level headings might be good for us to implement.
22. when following a link to another page, the position on the screen on the previous page affects where one lands - for example, if at the bottom of the screen (physical layout) when following a link, like one at the bottom of the page, then you land at of the screen on the linked page - so not quite at the top of that page.  This affects how you then interact with that page, not having arrived at the start!  Particularly obvious with the `login` link at the bottom of the article page, which then arrives on the press logo at the bottom of the login page.
23. navigation between pages - there are no breadcrumbs, no sitemap!
24. navigation within pages - a skip to main content would be nice and something that makes it clear one has reached the footer.   For long lists of articles, some way to know how long the list - either pagination or a 'showing results x of y would be helpful.
25. dates written as "YYYY-MM-DD" are not very readable.
26. clean bug - Where the news article is being truncated for the summary on the home page, if there is a line break this is not being translated into a space when concatenating with the next line, they are being run together.  See #4207

**Comment by StephDriver** [comment]
## Issues Identified:

1. #4209 - this is for WCAG 2.2 AAA, so not required in the current phase of the A11y project.  Issue outlined for consideration in the next phase.
2.

**Comment by StephDriver** [comment]
The results here are now integrated into the rest of the audit.
